userportaldatamodel/notification.py

```diff
@@ -0,0 +1,45 @@
+from . import Base
+import datetime
+from sqlalchemy import (
+    Integer,
+    String,
+    Column,
+    Table,
+    Boolean,
+    BigInteger,
+    DateTime,
+    Text,
+)
+from sqlalchemy import UniqueConstraint, Index, CheckConstraint
+from sqlalchemy.orm.collections import attribute_mapped_collection
+from sqlalchemy.ext.associationproxy import association_proxy
+from sqlalchemy.dialects.postgresql import ARRAY, JSONB
+from sqlalchemy.orm import relationship, backref
+from sqlalchemy.schema import ForeignKey
+from sqlalchemy.sql import func
+from sqlalchemy.types import LargeBinary
+from sqlalchemy.orm.collections import MappedCollection, collection
+import json
+
+from marshmallow_sqlalchemy import SQLAlchemyAutoSchema
+from marshmallow_sqlalchemy.fields import Nested
+
+class Notification(Base):
+    __tablename__ = "notification"
+
+    notification_id = Column(Integer, ForeignKey("notification_log.id"), primary_key=True, nullable=False) 
+    user_id = Column(Integer, ForeignKey("associated_user_roles.id"), primary_key=True, nullable=False)
```

Comment:
Also this should be a foreign key reference on the Associated_user table instead of associted_user_roles and the key should be user_id not id

Reply:
This should actually be just an integer as it is the user_id from fence. The associated_user table is populated only with users that started a project request, while the notification will be see by everyone.

Reply:
not a foreign key, change to integer @okmbaye

Reply:
ah ok so when Amanuensis adds rows to this table it will either have to send a request to fence.

Reply:
I was thinking more just using the logged in user information from the authutils package

Reply:
I agree that is definitely the better way of doing it